examples/oauth/src/pages/index.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  GoogleOAuthProvider,
+  GoogleLogin,
+  useGoogleLogin,
+} from "@react-oauth/google";
+import Image from "next/image";
+import styles from "./index.module.css";
+import { createActivityPoller, TurnkeyClient } from "@turnkey/http";
+import { IframeStamper } from "@turnkey/iframe-stamper";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+import * as React from "react";
+import { useState } from "react";
+import { Auth } from "@/components/Auth";
+import { sha256 } from "@noble/hashes/sha2";
+import { bytesToHex } from "@noble/hashes/utils";
+
+/**
+ * Type definition for the server response coming back from `/api/auth`
+ */
+type AuthResponse = {
+  userId: string;
+  apiKeyId: string;
+  credentialBundle: string;
+};
+
+/**
+ * Type definitions for the form data (client-side forms)
+ */
+type InjectCredentialsFormData = {
+  walletName: string;
+};
+type AuthFormData = {
+  invalidateExisting: boolean;
+};
+
+export default function AuthPage() {
+  const [authResponse, setAuthResponse] = useState<AuthResponse | null>(null);
+  const [iframeStamper, setIframeStamper] = useState<IframeStamper | null>(
+    null
+  );
+  const { register: authFormRegister, handleSubmit: authFormSubmit } =
+    useForm<AuthFormData>();
+  const {
+    register: injectCredentialsFormRegister,
+    handleSubmit: injectCredentialsFormSubmit,
+  } = useForm<InjectCredentialsFormData>();
+
+  const handleGoogleLogin = async (response: any) => {
+    let targetSubOrgId: string;
+    const getSuborgsResponse = await axios.post("api/getSuborgs", {
+      filterType: "OIDC_TOKEN",
+      filterValue: response.credential,
+    });
+    targetSubOrgId = getSuborgsResponse.data.organizationIds[0]; // If you don't have a 1:1 relationship for suborgs:oauthProviders you will need to manage this
```

Comment:
this is a good point; I think some a _little_ more detail could be super helpful!